.github/workflows/build.yml

```diff
@@ -0,0 +1,249 @@
+    # we only push tags with releases, and we handle releases explicitly
+    branches:
+      - '**'
+    tags-ignore:
+      - '**'
+
+  pull_request:
+
+  release:
+    types:
+      - 'published'
+
+jobs:
+
+  # run unit (and integration tests if account secrets available) on our build matrix
+  test:
+
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15, windows-2019]
+        python: [3.6, 3.7, 3.8]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: install-py-dependencies
+        shell: bash
+        run: |
+          pip install -e ".[boto3,pandas,pysftp,tests]"
+
+          # ensure that numpy c extensions are installed on windows
+          # https://stackoverflow.com/a/59346525
+          if [ "${{startsWith(runner.os, 'Windows')}}" == "true" ]; then
+            pip uninstall -y numpy
+            pip uninstall -y setuptools
+            pip install setuptools
+            pip install numpy
+          fi
+
+      - name: run-unit-tests
+        shell: bash
+        run: |
+          pytest -sv tests/unit
+
+      # run integration tests iff the decryption keys for the test configuration are available.
+      # they will not be available in pull requests from forks.
+      - name: run-integration-tests
+        shell: bash
+        run: |
+          if [ -z "${{ secrets.encrypted_d17283647768_key }}" ]  || [ -z "${{ secrets.encrypted_d17283647768_key }}" ]; then
+            echo "No test configuration decryption keys available, skipping integration tests"
+
+          else
+            # host used in pysftp tests
+            mkdir -p ~/.ssh
+            ssh-keyscan -H ec2-18-209-45-78.compute-1.amazonaws.com >> ~/.ssh/known_hosts
+
+            # decrypt the encrypted test synapse configuration
+            openssl aes-256-cbc -K ${{ secrets.encrypted_d17283647768_key }} -iv ${{ secrets.encrypted_d17283647768_iv }} -in test.synapseConfig.enc -out test.synapseConfig -d
+            mv test.synapseConfig ~/.synapseConfig
+
+            pytest -sv tests/integration
+          fi
+
+  # on a GitHub release, build the pip package and upload it to the appropriate pypi server.
+  # we upload prereleases to test.pypi.org and releases to pypi.org.
+  # the package is also uploaded to the GitHub release as an asset.
+  package-deploy:
+    needs: test
+
+    runs-on: ubuntu-18.04
+
+    if: github.event_name == 'release'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+
+      - name: set-release-env
+        shell: bash
+        run: |
+          RELEASE_TAG="${{ github.event.release.tag_name }}"
+          if [[ $RELEASE_TAG =~ ^v?([[:digit:]\.]+)(-rc)? ]]; then
+            VERSION="${BASH_REMATCH[1]}"
+            if [[ "${{ github.event.release.prerelease}}" == "true" ]]; then
```

Comment:
if -rc is found and not pre-release fail the build

Reply:
Added a check confirming that -rc is present on prerelease/test deployments.